bytecomp/dll.ml

```diff
@@ -132,32 +132,15 @@ let ld_conf_contents () =
 
 (* Split the CAML_LD_LIBRARY_PATH environment variable and return
    the corresponding list of directories.  *)
-
-let split str sep =
-  let rec split_rec pos =
-    if pos >= String.length str then [] else begin
```

Comment:
Note that you replaced this function with `String.split_on_char`, which never returns an empty list. This will make a difference when the environment variable is present but empty. In that case, it amounts to adding the current directory to the search path.

It may be OK to do that, but it's a subtle change that should be carefully considered.

Reply:
Good catch!

Even if it is a desirable change, better to consider that in a separate PR. I will amend this one to restore the old behaviour.

Reply:
This is now fixed.